mdpow/analysis/ensemble.py

```diff
@@ -0,0 +1,499 @@
+            self._num_systems += 1
+            return
+        if isinstance(trajectory, list):
+            # setting abs paths for trajectory
+            trajectory = [os.path.abspath(trj) for trj in trajectory]
+        try:
+            self._ensemble[key] = mda.Universe(os.path.abspath(topology), trajectory)
+        except (ValueError, FileFormatWarning, NoDataError, MissingDataWarning, OSError) as err:
+            logger.warning('Error loading systems at %r', key)
+            raise NoDataWarning
+        else:
+            self._keys.append(key)
+            self._num_systems += 1
+            return
+
+    def pop(self, key):
+        """Removes and returns system at specified key.
+
+        Logs if KeyError is raised."""
+        system = self._ensemble.pop(key)
+        self._num_systems -= 1
+        return system
+
+    def select_atoms(self, *args, **kwargs):
+        """Returns :class:`~mdpow.analysis.ensemble.EnsembleAtomGroup` for
+         Universes from the :class:`~mdpow.analysis.ensemble.Ensemble`
+
+        Uses the same
+        `selection commands <https://docs.mdanalysis.org/stable/documentation_pages/selections.html>`_
+        as MDAnalysis."""
+        selections = {}
+        for key in self.keys():
+            try:
+                ag = self[key].select_atoms(*args, **kwargs)
+            except SelectionError as err:
+                logger.error("%r on system %r with selection settings %r %r", err, key, args, kwargs)
+                raise
+            else:
+                selections[key] = ag
+        return EnsembleAtomGroup(selections, ensemble_dir=self._ensemble_dir)
+
+    def select_systems(self, keys=None, solvents=None, interactions=None,
+                       lambdas=None, lambda_range=None):
+        """Select specific subset of systems and returns them in an Ensemble.
+
+        This can be accomplished in two ways, by specific keys, or by
+        specifying the desired system attributes solvents, interactions and
+        lambdas. All arguments are stored in list form.
+
+        Specific key workflow example::
+
+            Ens = Ensemble(dirname='Mol')
+            w_v_0_25 = Ens.select_systems(keys=[('water', 'VDW', '0000'),
+                                                ('water', 'VDW', '0025')]
+
+        For the system attributes workflow there are two ways of selecting lambdas,
+        the :param lambdas: keyword saves specific lambdas or the :param lambda_range:
+        which saves the lambdas that fall within the given range.
+
+        Specific lambdas example::
+
+            Ens = Ensemble(dirname='Mol')
+            w_v_0_25 = Ens.select_systems(solvents=['water'], interactions=['VDW'],
+                                          lambdas=['0000', '0025'])
+        Range of lambdas example::
+
+            Ens = Ensemble(dirname='Mol')
+            w_v = Ens.select_systems(solvents=['water'], interactions=['VDW'],
+                                     lambda_range=[0, 1])
```

Comment:
lambda_range is fragile (because of our simplistic naming convention) but convenient. Just a comment — let's keep it for MDPOW.